diff --git a/vcd/datasource_not_found_test.go b/vcd/datasource_not_found_test.go
index 4dcf7d2b..45ae0c66 100644
--- a/vcd/datasource_not_found_test.go
+++ b/vcd/datasource_not_found_test.go
@@ -185,7 +185,7 @@ func addMandatoryParams(dataSourceName string, mandatoryFields []string, t *test

                if (dataSourceName == "vcd_nsxt_edgegateway_bgp_configuration" || dataSourceName == "vcd_nsxt_alb_settings" ||
                        dataSourceName == "vcd_nsxt_firewall" || dataSourceName == "vcd_nsxt_route_advertisement" ||
-                       dataSourceName == "vcd_nsxt_edgegateway_rate_limiting") &&
+                       dataSourceName == "vcd_nsxt_edgegateway_rate_limiting" || dataSourceName == "vcd_nsxt_edgegateway_dhcp_forwarding") &&
                        mandatoryFields[fieldIndex] == "edge_gateway_id" {
                        // injecting fake Edge Gateway ID
                        templateFields = templateFields + `edge_gateway_id = "urn:vcloud:gateway:784feb3d-87e4-4905-202a-bfe9faa5476f"` + "\n"